frontend/src/modules/Map/MapView.tsx

```diff
@@ -22,41 +22,48 @@ export function MapView({ moduleContext }: ModuleFCProps<MapState>) {
+                id="deckgl" 
+                layers={[   {
+                    "@@type": "MapLayer",
+                    "id": "mesh-layer",
+                    "meshData": JSON.parse(surfDataQuery.data?.mesh_data),
+                    "frame": {
+                      "origin": [
+                        surfDataQuery.data?.x_ori,
+                        surfDataQuery.data?.y_ori
+                      ],
+                      "count": [
+                        surfDataQuery.data?.x_count,
+                        surfDataQuery.data?.y_count
+                      ],
+                      "increment": [
+                        surfDataQuery.data?.x_inc,
+                        surfDataQuery.data?.y_inc
+                      ],
+                      "rotDeg": surfDataQuery.data?.rot_deg,
+                      
+                    },
+                    
+                    "contours": [
+                      0,
+                      100
+                    ],
+                    "isContoursDepth": true,
+                    "gridLines": false,
+                    "material": true,
+                    "smoothShading": true,
+                    "colorMapName": "Physics"
+                  }, 
+                ]}
+            />
+           
 
-    return (
-        <div className="w-full h-full flex flex-col">
-            <div>
-                addressType: {surfAddr?.addressType ?? "---"}
-                <br />
-                caseUuid: {surfAddr?.caseUuid ?? "---"}
-                <br />
-                ensembleName: {surfAddr?.ensemble ?? "---"}
-                <br />
-                surfaceName: {surfAddr?.name ?? "---"}
-                <br />
-                surfaceAttribute: {surfAddr?.attribute ?? "---"}
-                <br />
-                timeOrInterval: {surfAddrAsAny?.timeOrInterval ?? "---"}
-                <br />
-                realizationNum: {surfAddrAsAny?.realizationNum ?? "---"}
-                <br />
-                statisticFunction: {surfAddrAsAny?.statisticFunction ?? "Single realization"}
-            </div>
-            <br />
-            min x,y: {surfDataQuery.data?.x_min.toFixed(2)}, {surfDataQuery.data?.y_min.toFixed(2)}
-            <br />
-            min/max val: {surfDataQuery.data?.val_min.toFixed(2)}, {surfDataQuery.data?.val_max.toFixed(2)}
-            <br />
-            rot: {surfDataQuery.data?.rot_deg.toFixed(2)}
-            <br />
-            renderCount: {renderCount.current}
-            <div className="flex-grow h-0" ref={wrapperDivRef}>
-                <img
-                    alt={surfDataQuery.status}
-                    src={surfDataQuery.data ? `data:image/png;base64,${surfDataQuery.data.base64_encoded_image}` : ""}
-                    style={{ objectFit: "contain", width: wrapperDivSize.width, height: wrapperDivSize.height }}
-                ></img>
-            </div>
         </div>
     );
+            }
+    return (<div/>)
```

Comment:
Is this correct? I don't see the opening `<div>` this is closing. Maybe it simply can return `null` (nothing)? Maybe even possible to remove the `return` statement?

Reply:
it works, but it is not correct no. The second return statement should be removed and the first should be wrapped in a "ApiStateWrapper". I couldn't get it to work though as DeckGL raised an exception due to invalid data.
Have to discuss with @rubenthoms